src/metafile.c

```diff
@@ -870,10 +872,13 @@ GpStatus
 gdip_metafile_clone (GpMetafile *metafile, GpMetafile **clonedmetafile)
 {
 	GpMetafile *mf = gdip_metafile_create ();
+	GpImage *base;
 	if (!mf)
 		return OutOfMemory;
 
-	memcpy (&mf->base.image_format, &metafile->base.image_format, sizeof (GUID));
+	gdip_bitmap_clone (&metafile->base, &base);
```

Comment:
do we need `gdip_bitmap_setactive(*cloneImage, NULL, 0);` like  in GdipCloneImage () as well, or should we just call that method instead?

we should probably also assert that `metafile->base->type == ImageTypeBitmap`.

Reply:
The base type actually isn’t Bitmap, it’s metafile, so this assertion would be invalid.